### PR TITLE
Enable uncertainty quantification using quantile loss

### PR DESCRIPTION
**Pull request by kailingding**
## Summary
- `torchts`
    - added `quantile_loss`
    - added `criterion_args` parameter in `TimeSeriesModel` class in order to enable users to pass in customized quantile
    - added unit test for `loss.py`
- `examples`
    - added an example notebook that shows how to use torchts to make prediction intervals
    
## Reviewer Request
@klane 

## Paper
https://dl.acm.org/doi/pdf/10.1145/3447548.3467325

### CODE REVIEW

**Comment by yuqirose** [comment]
@DongxiaW, pls review this commit.

File: torchts/nn/model.py [hunk 1]
```diff
@@ -77,7 +84,10 @@ def _step(self, batch, batch_idx, num_batches):
             y = self.scaler.inverse_transform(y)
             pred = self.scaler.inverse_transform(pred)
 
-        loss = self.criterion(pred, y)
+        if self.criterion_args is not None:
+            loss = self.criterion(pred, y, **self.criterion_args)
```
[the remark this answers is not in the document]
Comment (klane) [reply torchts/nn/model.py:88]: Added in 900df93.

**Review by klane — APPROVED** [review]
